### PR TITLE
Fix uncertain Ajax session termination outcomes

### PR DESCRIPTION
**Pull request by aavdberg**
## Summary

Models a `KILL_SESSIONS` (`0x42`) request whose write completed but whose `0x41` confirmation was lost as an uncertain outcome, rather than a confirmed failure. The coordinator waits for its managed HTS connection to recover, performs exactly one read-only `0x40` session-list verification, and reports success only when the target is absent. A target still present is reported as a confirmed non-termination; unavailable verification reports an explicit outcome-unknown error and tells the user not to retry automatically.

Bulk termination preserves already confirmed IDs and counts an in-flight target only when that one verification proves it disappeared. It never retransmits `0x42`.

Relates to #330.

## Test plan

- [ ] `make check` passes locally on a freshly built Docker image
- [x] New behaviour covered by unit tests in `tests/unit/`
- [x] Coverage stays at or above the 80% threshold
- [ ] User-facing strings updated in `strings.json` and the 14 translation files
- [ ] `README.md` / `CHANGELOG.md` updated when the change is user-visible

Focused tests cover a pre-send failure, post-send timeout, post-send connection close, confirmed absence/presence, verification failure, and the bulk partial-count path.

## Notes for the reviewer

The focused suite passes (128 tests), along with Ruff lint and format checks for the changed files. The full unit run completed 2,430 tests but has seven pre-existing failures in `tests/unit/test_notification.py` FCM-log assertions, unrelated to this branch. The active Home Assistant test container was updated and restarted, but no live destructive action was run because the prior disposable Desktop session had already been removed and confirmation was unavailable to choose from the remaining sessions.

### CODE REVIEW

**Comment by aavdberg** [comment]
Live validation completed successfully on `ha-skyair-test` after an explicitly approved, newly logged-in Ajax Desktop session was available.

The pre-action `0x40` list contained the selected Ajax Desktop session on Windows 11 Version 26H2 plus one Android session. `terminate_client_session` was called exactly once with `confirm: true`; HTS reconnected during the action. The one final read-only list contained only the Android session, so the selected Desktop session was gone and no automation error was recorded. This confirms the recovery path handles the previously observed close/confirmation-loss scenario without retrying the destructive `0x42` request.

The temporary one-shot test automation was removed immediately and the test container was restarted with `automations.yaml` restored to `[]`.

File: tests/unit/test_services.py [hunk 1]
```diff
@@ -177,6 +177,94 @@ async def test_terminate_other_sessions_excludes_current_session(self) -> None:
...
+
+    @pytest.mark.asyncio
+    async def test_uncertain_outcome_verification_lists_once_after_reconnect(self) -> None:
+        from custom_components.aegis_ajax.coordinator import AjaxCobrandedCoordinator
+
+        coordinator = object.__new__(AjaxCobrandedCoordinator)
+        coordinator._hts_client = MagicMock(is_connected=True)
+        coordinator._hts_client.get_client_sessions = AsyncMock(return_value=[self._session(1)])
+        coordinator._maybe_restart_hts = AsyncMock()
+
```
[the remark this answers is not in the document]
Comment (aavdberg) [reply tests/unit/test_services.py:251]: Good catch. The test now begins with a disconnected HTS client, and its `_maybe_restart_hts` mock switches it to connected on the first recovery attempt. This exercises the recovery branch without sleeping, then asserts the single read-only list request. Included in e1e7f81; `tests/unit/test_services.py` and Ruff checks pass.

**Comment by bvis** [comment]
@aavdberg — the shape of this is right and it is the correct problem to attack. A write that completed with a lost confirmation is genuinely not a failure, never retransmitting `0x42` is the only safe rule, and preserving the already-confirmed IDs so a partial bulk call cannot look like a call that did nothing is exactly the honesty the endpoint's rate limiting demands. Two blocking findings, both in `_async_verify_termination_after_uncertain_outcome`, and I am flagging them with numbers rather than opinions because I reproduced both.

## 1. The verification loop removes the reconnect throttle

`_run_hts_lifecycle` has no backoff, by design — on failure it logs "The integration will retry on the next poll cycle" and clears `_hts_client`. The poll cycle *is* the throttle. `_maybe_restart_hts` was only ever called from `_async_update_data`, minutes apart.

Calling it from a `while` loop with `await asyncio.sleep(0.1)` removes that. When a connect fails fast, each iteration finds `_hts_client is None` and starts a fresh HTS connect: TCP, TLS and the auth handshake against Ajax. I measured it by driving the real `_maybe_restart_hts` with a `_start_hts` that fails the way `_run_hts_lifecycle` does:

```python
coordinator._hts_task = None
coordinator._hts_client = None
starts = 0
async def fake_start_hts() -> None:
    nonlocal starts
    starts += 1
    coordinator._hts_client = None   # what _run_hts_lifecycle does on failure
coordinator._start_hts = AsyncMock(side_effect=fake_start_hts)

ticks = iter([i * 0.1 for i in range(0, 400)])
with (
    patch("custom_components.aegis_ajax.coordinator.time.monotonic", lambda: next(ticks)),
    patch("asyncio.sleep", AsyncMock()),
    pytest.raises(HtsConnectionError, match="did not reconnect"),
):
    await coordinator._async_verify_termination_after_uncertain_outcome(2)
print(starts)   # 299
```

**299 connection attempts inside the 30 s window.** A slower-failing connect gives fewer, but even a one-second failure is ~30 attempts where the design allows one. That is the invariant in `CLAUDE.md`: the integration must never call Ajax more than the app would, and a reconnect storm on a code path like this is the kind of thing that gets an integration banned for everyone at once.

`test_uncertain_outcome_verification_lists_once_after_reconnect` cannot see this, because its `_maybe_restart_hts` mock flips `is_connected` to `True` on the first call. It proves one list request in the happy path, which is worth having, but it is not a bound on the failure path.

**Suggested fix:** call `_maybe_restart_hts()` once, before the wait loop, then poll only `is_connected` inside it. If that single attempt does not come up within the window, the existing "did not reconnect, do not retry automatically" error is the right and honest answer — the ordinary poll cycle will reconnect afterwards, and the user has an explicit outcome-unknown either way. If you would rather keep retrying, space the attempts (a 5 s interval is six attempts in the same window) and say so in the docstring.

## 2. An empty session list is read as proof of absence

```python
return all(session.session_id != session_id for session in sessions)
```

`all()` over an empty list is `True`, so an empty `0x41` list is reported as a confirmed termination:

```python
coordinator._hts_client.get_client_sessions = AsyncMock(return_value=[])
assert await coordinator._async_verify_termination_after_uncertain_outcome(2) is True  # passes
```

Two things make that reachable rather than theoretical. `_parse_client_sessions` returns `[]` instead of raising for a payload it cannot decode — deliberately, so the read-only service does not fail — which I confirmed directly: `HtsClient._parse_client_sessions([b"\x41", b"\xff"]) == []`. And `0x40` starts throttling after roughly seven requests, so a degraded or truncated answer is a live possibility on precisely this path, moments after a connection drop.

The list also can never legitimately be empty: the account always holds at least the integration's own session, which is why the `is_self_identity` refusal can rely on finding it. So `[]` always means the read failed, and right now a failed read is reported to the user as a successful termination — the one outcome this PR exists to prevent.

**Suggested fix:** treat a list that does not contain our own session as a failed verification and raise the existing outcome-unknown error. That is strictly better than an emptiness check, because it also catches a truncated list that happens to be non-empty.

## Everything else

The rest reads well. `HtsTerminationOutcomeUnknownError` subclassing `HtsConnectionError` with the specific `except` clause ordered first in both `kill_client_sessions` and `_send_user_registration_request` is correct; the `request_sent` flag genuinely distinguishes pre-send from post-send; and `terminated = [*exc.succeeded_session_ids, exc.session_id]` counts the uncertain target only on a positive verification, which matches the summary.

Once those two are addressed I will merge. Please add tests for both failure paths — a never-reconnecting HTS and an empty or self-less list — since those are the paths that carry the risk, and thank you for the live validation notes on #330; they made the review much faster.

**Comment by aavdberg** [comment]
Thanks — both reports were actionable, and the reconnect storm is fixed in 4b93057.

1. `_maybe_restart_hts()` now runs exactly once before the bounded wait. The loop then observes only the lifecycle-owned client until it connects or the 30-second deadline expires. The new no-reconnect test advances monotonic time straight to the deadline and asserts a single restart attempt, with no real sleep.

2. An empty post-recovery session list now raises the explicit outcome-unknown error rather than proving absence. The added test pins that behavior.

I did not add the suggested `is_self_identity` requirement. The immediately preceding successful live validation on `ha-skyair-test` returned two normal account sessions (the selected Ajax Desktop session and an Android client) and no `0x07` / `is_self_identity` record, yet a subsequent final read proved the Desktop record absent. Requiring an identity record would therefore reject the verified live server shape. The guard now treats the demonstrably malformed empty response as unknown; a non-empty parsed list remains the protocol response that the current service can verify against.

Focused session tests (130) and Ruff checks pass.

**Review by bvis — APPROVED** [review]
Both findings are fixed, and I checked them by mutation rather than by reading the diff, because that is what the first round taught me about this path.

**The reconnect storm is gone and the test genuinely bounds it.** `_maybe_restart_hts()` now runs once before the bounded wait and the loop observes only `is_connected`. To be sure the new test is a real bound and not a coincidence of the monotonic stub, I applied the exact mutation that reintroduces the defect — kept your pre-loop call *and* added one inside the loop — and `test_uncertain_outcome_verification_lists_once_after_reconnect` fails on it. One attempt per user action, which is the right owner for the rate limit on a service call. Same for the empty-list guard: removing the `if not sessions` block fails `test_uncertain_outcome_empty_verification_is_unknown`.

**You were right to refuse the `is_self_identity` requirement, and the reason matters more than the refusal.** My claim rested on the account always exposing its own session as a `0x07` record; your live read on a real account returned two sessions carrying none, and that is decisive — the guard I asked for would have made verification fail permanently on that server shape, turning a fix into a regression for exactly the users who need it. The guard you shipped covers the case I could actually prove reachable, which is the parser returning `[]` for a payload it cannot decode. Thank you for pushing back with a capture instead of implementing it.

One residual gap for the record, not for this PR: a verification list that is non-empty but truncated exactly at the target still reads as a confirmed termination, because `_parse_client_sessions` drops a trailing incomplete record by design. It is much narrower than what you fixed, and closing it needs corroboration against the pre-kill list rather than an identity record. I will note it on #330 as a follow-up.

Full unit suite green (2,438), all ten CI checks green, and the API delta is what the body states: one extra read-only `0x40` on the uncertain path only, no `0x42` ever retransmitted. Merging. It ships in the next `1.19.0` beta.
